packages/api/portfolioDrafts/middyHandler.ts

```diff
@@ -0,0 +1,98 @@
+import middy, { MiddyfiedHandler } from "@middy/core";
+import { APIGatewayProxyEvent, APIGatewayProxyResult, Handler, Context } from "aws-lambda";
+import jsonBodyParser from "@middy/http-json-body-parser";
+import validator from "@middy/validator";
+import { PortfolioStep, schema, schemaWrapper } from "../models/PortfolioStep";
+import { ApiSuccessResponse, DatabaseResult, DynamoDBException, SuccessStatusCode } from "../utils/response";
+import { NO_SUCH_PORTFOLIO_DRAFT } from "../utils/errors";
+import { PORTFOLIO_STEP } from "../models/PortfolioDraft";
+import { UpdateCommand } from "@aws-sdk/lib-dynamodb";
+import { dynamodbDocumentClient as client } from "../utils/dynamodb";
+import JSONErrorHandlerMiddleware from "middy-middleware-json-error-handler";
+import httpEventNormalizer from "@middy/http-event-normalizer";
+import { customMiddleware } from "../utils/customMiddleware";
+
+// This is a wrapper for the ApiGatewayProxyEvent will be moved somewhere else
+// middy validator & http-json-body-parser gives us a parsed, validated body in the APIGatewayProxyEvent with type of <T>
+// middy http-event-normalizer makes several fields on our ValidatedEvent NonNullable (see portfolioDraftId!, might need tweaking)
+interface NormalizedValidatedEvent<T> extends Omit<APIGatewayProxyEvent, "body"> {
+  queryStringParameters: NonNullable<APIGatewayProxyEvent["queryStringParameters"]>;
+  multiValueQueryStringParameters: NonNullable<APIGatewayProxyEvent["multiValueQueryStringParameters"]>;
+  pathParameters: NonNullable<APIGatewayProxyEvent["pathParameters"]>;
+  body: T; // parsed body is type T
+}
```

Comment:
Can you explain why it's necessary to use `Omit` for `body` but not for the rest of these?